### PR TITLE
Update Compose to 1.2.0, Kotlin to 1.7.0, and target SDK to 32

### PR DESCRIPTION
**Pull request by skydoves**
### 🎯 Goal

Update Compose to 1.2.0, Kotlin to 1.7.0, and target SDK to 32. Also updated dependencies below along with the Compose 1.2.0 and Kotlin 1.7.0.

- accompanist `0.25.0` stable.
- Dokka `1.7.0`.
- KSP `1.7.0`.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

gif

### CODE REVIEW

**Comment by filbabic** [comment]
Should we also cover this since we're pushing the Kotlin version up: https://github.com/GetStream/stream-chat-android/issues/3238

**Comment by filbabic** [comment]
@adasiewiczr Compose and Kotlin have to go hand-in-hand - in this case we're updating to 1.7.0 due to Compose 1.2.0, do we think it's okay, are we okay with the upgrade without a breaking change?

**Review by filbabic — APPROVED** [review]
From my standpoint, everything looks and works good! :]

I couldn't find any issues and the code updates are great!

The only concern I had was pushing to 1.7.0 of Kotlin, but we have to do that regardless because of Compose. I'm not sure if it's a breaking change for our users and how it'll work compatibility-wise.

But I think there were no major API or functionality changes between 1.6.21 and 1.7

File: buildSrc/src/main/kotlin/io/getstream/chat/android/Configuration.kt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 package io.getstream.chat.android
 
 object Configuration {
-    const val compileSdk = 31
-    const val targetSdk = 31
+    const val compileSdk = 32
+    const val targetSdk = 32
```
Comment (adasiewiczr) [inline buildSrc/src/main/kotlin/io/getstream/chat/android/Configuration.kt:5]: We shouldn't bump compile and target SDK versions for now as it's a breaking change and we are not ready to do that as we didn't adapt to new permissions system, etc.

**Review by adasiewiczr — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/kotlin/io/getstream/chat/android/Configuration.kt (hunk 1 above)
Comment (JcMinarro) [reply buildSrc/src/main/kotlin/io/getstream/chat/android/Configuration.kt:5]: I was concern about that as well, but reviewing documentation [SDK 32 is Android 12L](https://developer.android.com/studio/releases/platforms#12), the new permissions system is added on [SDK 33 (Android 13)](https://developer.android.com/about/versions/13/behavior-changes-13)

Comment (adasiewiczr) [reply buildSrc/src/main/kotlin/io/getstream/chat/android/Configuration.kt:5]: @JcMinarro nice catch! We should be good to bump both then

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**